backends/azure/azure.go

```diff
@@ -0,0 +1,482 @@
+	Container string `yaml:"container"`
+
+	// CreateBucket tells us to try to create the bucket
+	CreateContainer bool `yaml:"create_container"`
+
+	// GlobalPrefix is a prefix applied to all operations, allowing work within a prefix
+	// seamlessly
+	GlobalPrefix string `yaml:"global_prefix"`
+
+	// EndpointURL can be set to something like "http://localhost:9000" for local testing
+	EndpointURL string `yaml:"endpoint_url"`
+
+	// DisableContentMd5 defines whether to disable sending the Content-MD5 header
+	DisableContentMd5 bool `yaml:"disable_send_content_md5"`
+
+	// TLS allows customising the TLS configuration
+	// See https://github.com/PowerDNS/go-tlsconfig for the available options
+	TLS tlsconfig.Config `yaml:"tls"`
+
+	// InitTimeout is the time we allow for initialisation, like credential
+	// checking and bucket creation. It defaults to DefaultInitTimeout, which
+	// is currently 20s.
+	InitTimeout time.Duration `yaml:"init_timeout"`
+
+	// UseUpdateMarker makes the backend write and read a file to determine if
+	// it can cache the last List command. The file contains the name of the
+	// last file stored or deleted.
+	// This can reduce the number of LIST commands sent to Azure, replacing them
+	// with GET commands that are about 12x cheaper.
+	// If enabled, it MUST be enabled on all instances!
+	// CAVEAT: This will NOT work correctly if the bucket itself is replicated
+	//         in an active-active fashion between data centers! In that case
+	//         do not enable this option.
+	UseUpdateMarker bool `yaml:"use_update_marker"`
+	// UpdateMarkerForceListInterval is used when UseUpdateMarker is enabled.
+	// A LIST command will be sent when this interval has passed without a
+	// change in marker, to ensure a full sync even if the marker would for
+	// some reason get out of sync.
+	UpdateMarkerForceListInterval time.Duration `yaml:"update_marker_force_list_interval"`
+
+	// Concurrency defines the max number of concurrent uploads to be performed to upload the file.
+	// Each concurrent upload will create a buffer of size BlockSize.  The default value is one.
+	// https://github.com/Azure/azure-sdk-for-go/blob/e5c902ce7aca5aa0f4c7bb7e46c18c8fc91ad458/sdk/storage/azblob/blockblob/models.go#L264
+	Concurrency int `yaml:"concurrency"`
+
+	// Not loaded from YAML
+	Logger logr.Logger `yaml:"-"`
+}
+
+type Backend struct {
+	opt        Options
+	client     *azblob.Client
+	log        logr.Logger
+	markerName string
+
+	mu         sync.Mutex
+	lastMarker string
+	lastList   simpleblob.BlobList
+	lastTime   time.Time
+}
+
+func (o Options) Check() error {
+	if o.UseEnvCreds {
+		return nil
+	}
+
+	hasSecretsCreds := o.AccountName != "" && o.AccountKey != ""
+
+	if !hasSecretsCreds {
+		return fmt.Errorf("azure storage.options: account_name and account_key are required")
+	}
+
+	if o.Container == "" {
+		return fmt.Errorf("azure storage.options: container is required")
+	}
```

Comment:
With the `(Options).UseEnvCreds` short-circuit above, this check isn't happening in that case. Is that intentional?